beacon-chain/rpc/prysm/v1alpha1/beacon/beacon_test.go

```diff
@@ -11,7 +12,10 @@ func TestMain(m *testing.M) {
 	// Use minimal config to reduce test setup time.
 	prevConfig := params.BeaconConfig().Copy()
 	defer params.OverrideBeaconConfig(prevConfig)
-	params.OverrideBeaconConfig(params.MinimalSpecConfig())
+	cfg := params.MinimalSpecConfig()
+	cfg.EpochsPerHistoricalVector = customtypes.RandaoMixesSize
+	cfg.SlotsPerHistoricalRoot = customtypes.BlockRootsSize
+	params.OverrideBeaconConfig(cfg)
```

Comment:
This and similar changes are required because using the default minimal config will not work when tests are ran with mainnet configuration. In such case the minimal config's values for array sizes will not match values in the beacon chain which are taken from constants.

Reply:
Wouldnt we have the same issue with our proto state ? As that is also built at compile time via constant

Reply:
It's because of the changes here: https://github.com/prysmaticlabs/prysm/pull/9820/files#diff-d937da9a3b4bd7fb257ce341af7995d11427a0460542de503632824818132449

Size of roots generated in `OptimizedGenesisBeaconState` depended on config, so they always aligned with the overridden config. Now they rely on constants, which means the config values also have to be equal to constants.

Reply:
We should then cleanup this better and use bazel to run this via a minimal config. 
https://github.com/prysmaticlabs/prysm/blob/b4ba8956ff81c72b94d5e715b60ff007b8b0269c/beacon-chain/rpc/prysm/v1alpha1/beacon/BUILD.bazel#L81

Over here set the key as 
```
eth_network = "minimal",
```